2-mandatory/2-retrying-random-numbers.js

```diff
@@ -5,20 +5,26 @@
 
 // This function shouldn't be changed
 function generateRandomNumber() {
-    console.log("Generating number...");
-    return Math.round(Math.random() * 100);
+  console.log("Generating number...");
+  return Math.round(Math.random() * 100);
 }
 
 function getRandomNumberGreaterThan50() {
-    // TODO - implement using a do-while loop
+  // TODO - implement using a do-while loop
+  //let i = 0;
+  let greaterThan50 = 0;
+  do {
+    greaterThan50 = generateRandomNumber();
+  } while (greaterThan50 <= 50);
+  return greaterThan50;
```

Comment:
Yeah! You got the answer right. You can consider renaming the variable to something like "theGeneratedNum" etc sine it stores a number. theGeneratedNum <= 50 will reads more natural. The variable name **greaterThan50** is better suited for naming boolean variables with yes or no status.